### ISSUE TITLE
Regression on scope tracking

### ISSUE THREAD

**Issue by josemarluedke** [ember-template-lint/ember-template-lint#451]
I believe the changes in #450 caused a regression on tracking the scope of block params:

```
my-app/pods/components/application-nav/template.hbs
  3:4  error  'menu' is defined but never used  no-unused-block-params
```

Here is a reproduction.
```hbs
<BurgerMenu as |burger|>
  <burger.menu as |menu|>
    <header>
      Something
    </header>

    <menu.item>
      {{#link-to 'finder.bonds'}}
        {{fa-icon 'university'}} <span>Issuances</span>
      {{/link-to}}
    </menu.item>
  </burger.menu>

  <burger.outlet>
    {{yield (hash state=burger.state)}}
  </burger.outlet>
</BurgerMenu>
```

Note, the same regression happens if I call the components using curry braces.

**Comment by rwjblue** [comment]
@josemarluedke - Would you mind putting up a failing test PR (with tests for both angle bracket invocation and curly)?

**Comment by boris-petrov** [comment]
We also got a couple of false-positives:

```handlebars
{{#each sections as |section index|}}
  <li class="h3">{{section}}</li>
  <li class="center-block">
    {{#if (eq index 0)}}
    {{/if}}
  </li>
{{/each}}
```

`2:2  error  'index' is defined but never used  no-unused-block-params`

Not sure if this is the same case as the one above, just thought to mention it.

**Comment by rwjblue** [comment]
Yes, seems likely to be related.

**Comment by Serabe** [comment]
I debugged this a bit, but cannot find the culprit. Here is what I found:

1. The minimal failing test is `{{#my-modal as |modal|}}<h2>{{t "Some.Helper"}}</h2>{{display-item open=modal.isOpen}}{{/my-modal}}`.
2. I debugged the pushing and popping of frames, and it happens that, after closing `h2`, all the frames are popped and pushed again. Seem to happen again after `{{display-item}}` which causes any register of the variable being used disappear.

**Comment by rwjblue** [comment]
@Serabe - Thank you for digging in, your findings were very helpful in tracking it down.

#452 now includes a fix...
